### PR TITLE
first iteration to add a get for color

### PR DESCRIPTION
**Pull request by jtslear**
@drapergeek, can you assist me with the following error for my assertion testing?

```
% mix test
Compiled web/models/color.ex
Generated elixirlight app
Consolidated Poison.Encoder
.....

  1) test GET /api/color (Elixirlight.ColorControllerTest)
     test/controllers/color_controller_test.exs:11
     ** (FunctionClauseError) no function clause matching in Kernel.=~/2
     stacktrace:
       (elixir) lib/kernel.ex:1500: Kernel.=~(%{"color" => "#ff0000"}, "{\"color\":\"#ff0000\"}")
       test/controllers/color_controller_test.exs:14



Finished in 0.1 seconds (0.07s on load, 0.03s on tests)
6 tests, 1 failure

Randomized with seed 281561
```

### CODE REVIEW

**Comment by jtslear** [comment]
The end goal of this PR would be to resolve #3

File: test/controllers/color_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  test "GET /api/healthcheck", %{conn: conn} do
+    conn = get conn, "/api/healthcheck"
+    assert json_response(conn, 200)
+  end
+
+  test "GET /api/color", %{conn: conn} do
+    color_as_json = Poison.encode!(%{color: "#ff0000"})
+    conn = get conn, "/api/color"
+    assert json_response(conn, 200) =~ color_as_json
```
Comment (drapergeek) [inline test/controllers/color_controller_test.exs]: I don't think you can do this. I think you need two assertions here.

```
assert json_response(conn, 200)
assert conn.response_body =~ color_as_json
```

`json_response` doesn't return anything so the =~ is being called on `Kernel` it seems. https://hexdocs.pm/phoenix/Phoenix.ConnTest.html#json_response/2

Comment (drapergeek) [inline test/controllers/color_controller_test.exs]: Also, I've usually seen:

```
response = get conn, "/api/color"
```

Just so it is easy to differentiate that it is now a response object that is already been through the connection.

**Comment by jtslear** [comment]
So your tip worked well.  Now I'm stuck at a new spot.  In the test where I'm doing a POST, I cannot find the body of the request.  `params` is empty, and `conn` does not appear to contain a body that contains the data I'm sending to it.

**Comment by drapergeek** [comment]
Are you sure it is hitting the correct action. From other codebases I'm seeing this is how you would pass that in:

``` elixir

    conn = post conn, announcement_path(conn, :create), %{
      announcement: announcement_params,
      interest_names: ["foo"]
    }
```

I'm not too sure, just trying to offer some assistance.

**Comment by jtslear** [comment]
@drapergeek mind a final review before merge?

File: test/controllers/color_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defmodule Elixirlight.ColorControllerTest do
+  use Elixirlight.ConnCase
+
+  defstruct [:color]
+
+  test "GET / ensure color picker exists", %{conn: conn} do
+    conn = get conn, "/"
+    assert conn.resp_body =~ "value=\"#ff0000\""
```
Comment (drapergeek) [inline test/controllers/color_controller_test.exs]: You may want to use `html_response(conn, 200)` here.

**Comment by drapergeek** [comment]
LGTM :shipit:

**Pull request merged**